### DISCUSSION TITLE
Importing org.jooq.SQLDialect in 3.19.2

### DISCUSSION THREAD

**Discussion by littlebytesof** [jOOQ/jOOQ#16103] [Q&A] [answered]
When importing org.jooq.SQLDialect in a Kotlin SpringBoot project, my IDE(IntelliJ) complains of being unable to locate it:

<img width="1792" alt="1" src="https://github.com/jOOQ/jOOQ/assets/54659959/9c818d14-e9d0-4679-90bd-e13eca3832ac">

The jOOQ version employed here is 3.19.2, and this does not occur when using version 3.18.7.

The difference I found is that the SQLDialect class is defined in .class file(the IDE says that the source code has been recreated from a .class file) in 3.19.2, whereas it is in a .java file in 3.18.7.

The codes are compiled successfully in spite of the complaint.

**Comment by lukaseder** [comment, marked as answer]
Thanks for your message. I'm sorry, I don't know why your IDE might behave this way. I would just delete all `.idea` folders and other IDE meta data, clean the maven or gradle caches, and try again...

**Reply by littlebytesof** [reply]
Thank you for your response.
I tried cleaning the cache and restarting, but it didn't work.

Instead of integrating the jOOQ jar files of version 3.19.2 from the maven repository, I downloaded them from the site(https://www.jooq.org/download/) and integrated them locally:

<img width="1638" alt="1" src="https://github.com/jOOQ/jOOQ/assets/54659959/11250953-05f7-4b33-8f5d-02b333a877d2">

The warning has gone.
I'm still curious about what could have been the cause, but the problem has been resolved anyways.

**Reply by lukaseder** [reply]
The file is right there when I download the sources from Maven Central:

<img width="571" alt="image" src="https://github.com/jOOQ/jOOQ/assets/734593/30cf6cd2-f3a1-432c-8ce2-b22015617b95">

Sources:

- https://repo1.maven.org/maven2/org/jooq/jooq/3.19.2/jooq-3.19.2-sources.jar

So, I don't think it's a jOOQ problem.

**Reply by littlebytesof** [reply]
When I said "integrating the jOOQ jar files of version 3.19.2 from the maven repository", I meant doing it by calling 'implementation("org.jooq:jooq:3.9.12")' and others in "build.gradle.kts".

And in this case, what I find in the external libraries is the "SQLDialect.class" file instead of "SQLDialect.java":

<img width="1281" alt="1" src="https://github.com/jOOQ/jOOQ/assets/54659959/d7df803e-232f-4a5d-af74-26a810fb4561">

In agreement with your comment, I also believe that the IDE or some other factors, even my personal IDE settings, seem to be involved here. For your reference, I'd like to point out that this does not happen with the jOOQ 3.18.7 integrated in the identical manner.

I thank you for your support and prompt replies.
